test/http_server_spec/Rakefile

```
@@ -0,0 +1,65 @@
+def is_server_running?(port)
+  system("lsof -i:#{port}", out: '/dev/null')
+end
+
+def server_not_running_error(port)
+  STDERR.puts '~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~'
+  STDERR.puts "I tried to connect to the server on port #{port}."
+  STDERR.puts 'However, I was unable to successfully connect.'
+  STDERR.puts 'Maybe check if the server is up and running?'
+  STDERR.puts '~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~'
+  exit 1
+end
+
+desc "Run all of the tests in the suite"
+task :test do
+  port=4000
+  if is_server_running?(port)
+    puts `bundle exec spinach`
+  else
+    server_not_running_error(port)
+  end
+end
```

I like what you did here; I wonder if you could use a ruby constant?

```suggestion
PORT = 4000
...
desc "Run all of the tests in the suite"
task :test do
  if is_server_running?(PORT)
    puts `bundle exec spinach`
  else
    server_not_running_error(port)
  end
end
```

---

I like this much better! It's more Rubyish. I also do love the idea of attempting to refactor so that server is only in one place, because you are right it was unexpectedly challenging to figure out all the places where I needed to change port